src/main/scala/codacy/dockerApi/DockerEngine.scala

```diff
@@ -46,11 +46,18 @@ abstract class DockerEngine(Tool: Tool) {
         }))
 
         log("tool started")
-        Tool(
-          path = sourcePath,
-          conf = maybePatterns,
-          files = maybeFiles
-        )
+        try {
+          Tool.apply(
+            path = sourcePath,
+            conf = maybePatterns,
+            files = maybeFiles
+          )
+        } catch {
+          // We need to catch Throwable here to avoid JVM crashes
+          // Crashes can lead to docker not exiting properly
+          case e: Throwable =>
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fatal exception is being caught](https://www.codacy.com/app/Codacy/codacy-engine-scala-seed/file/2393792812/issues/source?bid=2836226&fileBranchId=3230242#l58)